lib/bpk-component-banner-alert/src/AnimateAndFade.js

```diff
@@ -82,7 +82,7 @@ class AnimateAndFade extends Component<Props> {
 
     const isCollapsed = this.props.animateOnEnter || !this.props.show;
 
-    this.shouldRenderHeight = true;
+    this.shouldRenderHeight = this.props.animateOnLeave;
```

Comment:
This line is pretty much the entire thing. I don't really understand this component, so basically these are the steps I followed to come to this conclusion:

---

I moved the stuff from `UNSAFE_componentWillReceiveProps` into `componentDidUpdate`. This worked fine for dismissable alerts but not expandable ones. 

So then I experimented with some of the props we were checking in `componentDidUpdate`, and after a bit of trial and error I discovered that changing the initial value of `shouldRenderHeight` to `false` made expandable alerts work correctly! 

But, **oh no**, now dismissable alerts weren't being removed correctly (the other alerts weren't sliding back up to cover the empty space it leaves).

So, my workaround was to make the initial value dependent on the `animateOnLeave` prop, as dismissable alerts set this to `true`.

This seems to work fine now. I think. This is a really complex component 😅